components/merkledb/src/db.rs

```diff
@@ -567,12 +521,24 @@ impl Snapshot for FlushedFork {
 }
 
 impl Fork {
+    /// Creates a fork based on the provided `patch`.
+    ///
+    /// Note: using created fork to modify data already present in `patch` may lead
+    /// to an inconsistent database state. Hence, this method is useful only if you
+    /// are sure that the fork and `patch` interacted with different indices.
+    pub fn from_patch(patch: Patch) -> Self {
```

Comment:
What do you think about implementing `From<Patch> for Fork`?

Reply:
I've changed this method to accept `snapshot` for consistency, therefore currently I can't implement `From<Patch`.

Reply:
@pavel-mukhanov Do we really need this method in case we implement `From<Patch` trait for `Fork` ? I suppose we can get rid of it at all, can't we ?